Add folsom stats to riak_search

russelldb opened the pull request:
Since there is no legacy output format for stats in search I'm using what I think is the correct format going forward, a 2 tuple of {app_name(), [stat()]}.

Just tested by attaching and running combinations of

```
riak_search_stat:update(index_begin).
riak_search_stat:update({index_end, N}).
riak_search_stat:update({index_entries, N}).
riak_search_stat:update(solr_query_begin).
riak_search_stat:update({solr_query_end, N}).
%% get stats
riak_search_stat:get_stats().
```

rzezeski committed:
Add various index/query stats

russelldb committed:
Add folsom stats to riak_search

rzezeski commented:
I just want to make a quick comment that I want to change the query stat name and catch all query paths but haven't had time to touch this yet.

buddhisthead commented:
Hey Ryan, I haven't reviewed that one yet. If you can summarize what
you want, I could maybe take a crack at making the changing if you
want. I can certainly wait on this one till end of day or even over
the weekend if you want to tweak it.

Cheers, Chris

On Thu, May 31, 2012 at 4:54 PM, Ryan Zezeski
reply@reply.github.com
wrote:

> I just want to make a quick comment that I want to change the query stat name and catch all query paths but haven't had time to touch this yet.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/basho/riak_search/pull/111#issuecomment-6050598

rzezeski committed:
Log stats for 3 unique search APIs

Riak Search currently has 3 unique search APIs.

1. Search for matching docs - `search`

2. Search as input to map/reduce - `search_fold`

3. Search and then fetch documents - `search_doc`

Since each does varying amounts of work it makes more sense to track
stats separate than together.  E.g. `search_doc` has potential to
perform much more work than `search` since it returns matching objects
and not just the keys.

russelldb commented:
So were do we stand on this now? I looked at Ryan's changes and they look good to me. Ryan, are you done? Can Chris review now?

rzezeski commented:
I just rebased, +1 to merge.
